lib/ansible/modules/cloud/vmware/vmware_host_powermgmt_policy.py

```diff
@@ -0,0 +1,210 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2018, Christian Kotte <christian.kotte@gmx.de>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = r'''
+---
+module: vmware_host_powermgmt_policy
+short_description: Manages the power management policy of an ESXI host system
+description:
+- This module can be used to manage the power management policy of ESXi host systems in given vCenter infrastructure.
+version_added: 2.8
+author:
+- Christian Kotte (@ckotte) <christian.kotte@gmx.de>
+requirements:
+- python >= 2.6
+- PyVmomi
+options:
+  policy:
+    description:
+    - Set the Power Management Policy of the host system.
+    choices: [ 'high-performance', 'balanced', 'low-power', 'custom' ]
+    default: 'balanced'
+  esxi_hostname:
+    description:
+    - Name of the host system to work with.
+    - This is required parameter if C(cluster_name) is not specified.
+  cluster_name:
+    description:
+    - Name of the cluster from which all host systems will be used.
+    - This is required parameter if C(esxi_hostname) is not specified.
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = r'''
+- name: Set the Power Management Policy of a host system to high-performance
+  vmware_host_powermgmt_policy:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    esxi_hostname: '{{ esxi_hostname }}'
+    policy: high-performance
+    validate_certs: no
+  delegate_to: localhost
+
+- name: Set the Power Management Policy of all host systems from cluster to high-performance
+  vmware_host_powermgmt_policy:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    cluster_name: '{{ cluster_name }}'
+    policy: high-performance
+    validate_certs: no
+  delegate_to: localhost
+'''
+
+RETURN = r'''
+result:
+    description: metadata about host system's power policy
+    returned: always
+    type: dict
+    sample: {
+        "esxi01": {
+            "msg": "power policy for host 'esxi01' already configured",
```

Comment:
I would actually return, per host:
* current_host_power_policy.name
* current_host_power_policy.short_name
* current_host_power_policy.description
* changed (true/false, to indicate if a change was made for the host specifically)

Reply:
The name and description parameters only return strings with "PowerPolicy.<shortName>.name" or "PowerPolicy.<shortName>.description". Not very useful to output.

Do you mean getting rid of msg entirely?

Reply:
He meant - 

return dictionary may / can contain

previous state of properties and current state of properties so user can distinguish between action.

Reply:
Would this be ok?
```
TASK [esxi : Configure Power Management Policy] **********************************************************************************************************************************************************************************************
ok: [esxi_host -> jump_server] => {"changed": true, "result": {"esxi_host": {"changed": true, "current_state": "high-performance", "desired_state": "high-performance", "error": "", "msg": "Power policy for host 'esxi_host' configured", "previous_state": "balanced"}}}
```
```
TASK [esxi : Configure Power Management Policy] **********************************************************************************************************************************************************************************************
ok: [esxi_host -> jump_server] => {"changed": false, "result": {"esxi_host": {"changed": false, "current_state": "high-performance", "desired_state": "high-performance", "error": "", "msg": "Power policy for host 'esxi_host' is already configured", "previous_state": "high-performance"}}}
```
I assume we should remove `msg` entirely or just use `Power policy configured` or `Power policy is already configured`.

Reply:
Perfect, we should keep msg as well with message.